src/scvi/data/_preprocessing.py

```diff
@@ -123,12 +123,18 @@ def poisson_gene_selection(
 
         # Calculate empirical statistics.
         sum_0 = np.asarray(data.sum(0)).ravel()
-        scaled_means = torch.from_numpy(sum_0 / sum_0.sum()).to(device)
         total_counts = torch.from_numpy(np.asarray(data.sum(1)).ravel()).to(device)
-
-        observed_fraction_zeros = torch.from_numpy(
-            np.asarray(1.0 - (data > 0).sum(0) / data.shape[0]).ravel()
-        ).to(device)
+        # in MPS we need to first change to float 32, as the MPS framework doesn't support float64.
+        if device.type == "mps":
```

Comment:
We could also do it for other devices. Float32 is sufficient for all computations.